test/instrumentations/mongodb/mongodb.test.ts

```diff
@@ -205,5 +210,76 @@ describe.each(versionsToTest(INSTRUMENTATION_NAME, INSTRUMENTATION_NAME))(
         );
       }
     );
+
+    itTest(
+      {
+        testName: `filter isMaster request: ${versionToTest}`,
+        packageName: INSTRUMENTATION_NAME,
+        version: versionToTest,
+        timeout: TEST_TIMEOUT,
+      },
+      async function () {
+        const exporterFile =  `${SPANS_DIR}/filter-isMaster@${versionToTest}.json`
+        let mongoConnectionUrl = getMongoContainerConnectionUrl(mongoContainer, versionToTest);
+        testApp = new TestApp(
+          TEST_APP_DIR,
+          INSTRUMENTATION_NAME,
+          {
+            spanDumpPath: exporterFile,
+            env: {
+              MONGODB_URL: mongoConnectionUrl.toString(),
+              OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT: '4096',
+            }
+          });
+
+        if (versionToTest.startsWith('3')) {
+
+          await testApp.invokeGetPath(`/mongodb-isMaster`);
+          // older versions of mongodb driver add extra spans
+          let spans = await testApp.getFinalSpans(2);
+          expect(getSpansByAttribute(spans, 'db.system', 'mongodb')).toHaveLength(1);
+        } else {
+          await testApp.invokeGetPath(`/mongodb-isMaster`);
+          let spans = await testApp.getFinalSpans(1);
+          expect(getSpansByAttribute(spans, 'db.system', 'mongodb')).toHaveLength(0);
+        }
```

Comment:
I usually try to avoid branching in tests. WDYT about this structure? 
```suggestion
        await testApp.invokeGetPath(`/mongodb-isMaster`);
        const expectedSpanCount = versionToTest.startsWith('3') ? 2 : 1;
        const expectedDbSystemAttributeSpans = versionToTest.startsWith('3') ? 1 : 0;
        
        const spans = await testApp.getFinalSpans(expectedSpanCount);
        expect(getSpansByAttribute(spans, 'db.system', 'mongodb')).toHaveLength(expectedDbSystemAttributeSpans);
```

Reply:
Done